source/fe/fe_system.cc

```diff
@@ -1204,8 +1118,28 @@ compute_fill (const Mapping<dim,spacedim>                      &mapping,
                           fe_data.update_flags);
 
 
-  if (flags & (update_values | update_gradients))
+  if (flags & (update_values | update_gradients | update_hessians))
     {
+      if (fe_data.is_first_cell())
+        {
+          // Initialize the internal FE objects for the base elements
+          for (unsigned int base_no=0; base_no<this->n_base_elements(); ++base_no)
```

Comment:
Yes, I think the "if (fe_data.is_first_cell())" block should be removed. I had pulled changes from upstream at commit 990a78a and then checked out the files I'd changed, it seems this may have overwritten #1315. I'll repeat the process at an earlier commit to ensure none of the relevant patches is overwritten.